python_modules/dagster/dagster/_core/definitions/partitioned_schedule.py

```diff
@@ -40,78 +29,35 @@ def build_schedule_from_partitioned_job(
-        partitions_def = cast(TimeWindowPartitionsDefinition, partitioned_config.partitions_def)
-    else:  # UnresolvedAssetJobDefinition
-        check.invariant(job.partitions_def is not None, "Job does not have a partitions_def.")
-        partition_set = cast(PartitionSetDefinition, job.get_partition_set_def())
-        partitions_def = cast(TimeWindowPartitionsDefinition, job.partitions_def)
-
-    check.inst(partitions_def, TimeWindowPartitionsDefinition)
-
-    minute_of_hour = cast(
-        int,
-        check.opt_int_param(minute_of_hour, "minute_of_hour", default=partitions_def.minute_offset),
-    )
-
-    if partitions_def.schedule_type == ScheduleType.HOURLY:
-        check.invariant(hour_of_day is None, "Cannot set hour parameter with hourly partitions.")
-
-    hour_of_day = cast(
-        int, check.opt_int_param(hour_of_day, "hour_of_day", default=partitions_def.hour_offset)
+    cron_schedule = partitions_def.get_cron_schedule(
+        minute_of_hour, hour_of_day, day_of_week, day_of_month
     )
-    execution_time = time(minute=minute_of_hour, hour=hour_of_day)
-
-    if partitions_def.schedule_type == ScheduleType.DAILY:
-        check.invariant(
-            day_of_week is None, "Cannot set day of week parameter with daily partitions."
-        )
-        check.invariant(
-            day_of_month is None, "Cannot set day of month parameter with daily partitions."
-        )
-
-    if partitions_def.schedule_type == ScheduleType.MONTHLY:
-        default = partitions_def.day_offset or 1
-        execution_day = check.opt_int_param(day_of_month, "day_of_month", default=default)
-    elif partitions_def.schedule_type == ScheduleType.WEEKLY:
-        default = partitions_def.day_offset or 0
-        execution_day = check.opt_int_param(day_of_week, "day_of_week", default=default)
-    else:
-        execution_day = 0
 
-    cron_schedule = get_cron_schedule(partitions_def.schedule_type, execution_time, execution_day)
-
-    schedule_def = partition_set.create_schedule_definition(
-        schedule_name=check.opt_str_param(name, "name", f"{job.name}_schedule"),
+    @schedule(
         cron_schedule=cron_schedule,
-        partition_selector=latest_window_partition_selector,
-        execution_timezone=partitions_def.timezone,
-        description=description,
         job=job,
         default_status=default_status,
+        execution_timezone=partitions_def.timezone,
+        name=check.opt_str_param(name, "name", f"{job.name}_schedule"),
+        description=check.opt_str_param(description, "description"),
     )
+    def schedule_def(context):
+        partition_keys = partitions_def.get_partition_keys(context.scheduled_execution_time)
+        if len(partition_keys) == 0:
+            return SkipReason()
 
-    return schedule_def
+        partition_key = partition_keys[-1]
 
+        yield job.run_request_for_partition(partition_key=partition_key, run_key=partition_key)
```

Comment:
is there guaranteed to be just one partition key that we want to execute over? Should we be yielding run requests for each partition?

Reply:
either way good to include a comment here as it's a bit confusing

Reply:
added